cedar-policy-core/src/entities/conformance.rs

```diff
@@ -176,6 +179,66 @@ pub fn typecheck_value_against_schematype(
     }
 }
 
+/// Check whether the given `RestrictedExpr` is a valid instance of `SchemaType`
+pub fn does_restricted_expr_implement_schematype(
+    expr: BorrowedRestrictedExpr<'_>,
+    expected_ty: &SchemaType,
+) -> bool {
+    use SchemaType::*;
+
+    match expected_ty {
+        Bool => expr.as_bool().is_some(),
+        Long => expr.as_long().is_some(),
+        String => expr.as_string().is_some(),
+        EmptySet => expr.as_set_elements().is_some(),
```

Comment:
Does this say that any set (even a non-empty set) implements the `EmptySet` type? I would expect that only empty sets implement `EmptySet.

Reply:
That matches the behavior in `is_consistent_with`
https://github.com/cedar-policy/cedar/blob/c1b078296da73b812dff28e88ccc1f7266d74c21/cedar-policy-core/src/entities/json/schema_types.rs#L123

Reply:
It's going to be right for `is_consistent_with` since that's documented as checking "here exist some concrete values that have both types", but I don't think it's going to be correct here. 

Also, it doesn't look like this match case is reachable.